src/wrappers/python/PyOpenEXR.cpp

```diff
@@ -0,0 +1,2755 @@
+        return npa;
+    }
+    
+    if (auto v = dynamic_cast<const PreviewImageAttribute*> (a))
+    {
+        auto I = v->value();
+        return py::cast(PyPreviewImage(I.width(), I.height(), I.pixels()));
+    }
+
+    if (auto v = dynamic_cast<const StringAttribute*> (a))
+    {
+        if (name == "type")
+        {
+            //
+            // The "type" attribute comes through as a string,
+            // but we want it to be the OpenEXR.Storage enum.
+            //
+                  
+            exr_storage_t t = EXR_STORAGE_LAST_TYPE;
+            if (v->value() == SCANLINEIMAGE) // "scanlineimage")
+                t = EXR_STORAGE_SCANLINE;
+            else if (v->value() == TILEDIMAGE) // "tiledimage")
+                t = EXR_STORAGE_TILED;
+            else if (v->value() == DEEPSCANLINE) // "deepscanline")
+                t = EXR_STORAGE_DEEP_SCANLINE;
+            else if (v->value() == DEEPTILE) // "deeptile") 
+                t = EXR_STORAGE_DEEP_TILED;
+            else
+                throw std::invalid_argument("unrecognized image 'type' attribute");
+            return py::cast(t);
+        }
+        return py::str(v->value());
+    }
+
+    if (auto v = dynamic_cast<const StringVectorAttribute*> (a))
+    {
+        auto l = py::list();
+        for (auto i = v->value().begin (); i != v->value().end(); i++)
+            l.append(py::str(*i));
+        return l;
+    }
+
+    if (auto v = dynamic_cast<const FloatVectorAttribute*> (a))
+    {
+        auto l = py::list();
+        for (auto i = v->value().begin(); i != v->value().end(); i++)
+            l.append(py::float_(*i));
+        return l;
+    }
+
+    if (auto v = dynamic_cast<const RationalAttribute*> (a))
+    {
+        py::module fractions = py::module::import("fractions");
+        py::object Fraction = fractions.attr("Fraction");
+        return Fraction(v->value().n, v->value().d);
+    }
+
+    if (auto v = dynamic_cast<const TileDescriptionAttribute*> (a))
+        return py::cast(v->value());
+
+    if (auto v = dynamic_cast<const TimeCodeAttribute*> (a))
+        return py::cast(v->value());
+
+    if (auto v = dynamic_cast<const V2iAttribute*> (a))
+        return make_v2(v->value());
+
+    if (auto v = dynamic_cast<const V2fAttribute*> (a))
+        return make_v2(v->value());
+
+    if (auto v = dynamic_cast<const V2dAttribute*> (a))
+        return make_v2(v->value());
+
+    if (auto v = dynamic_cast<const V3iAttribute*> (a))
+        return make_v3(v->value());
+    
+    if (auto v = dynamic_cast<const V3fAttribute*> (a))
+        return make_v3(v->value());
+    
+    if (auto v = dynamic_cast<const V3dAttribute*> (a))
+        return make_v3(v->value());
+    
+    std::stringstream err;
+    err << "unsupported attribute type: " << a->typeName();
+    throw std::runtime_error(err.str());
+    
+    return py::none();
+}
+    
+template <class P, class T>
+bool
+is_v2(const py::object& object, Vec2<T>& v)
```

Comment:
I can't find a copy of IMath's python bindings at hand, but should these marshallings better be in IMath rather than OpenEXR...?

Reply:
Also i understand this is internal, but this is actually a conversion rather than a read-only test, so maybe rename them to "convert_to_v2" or simply "to_v2"

Reply:
I renamed `is_v2()` to `objectToV2()` and the others similarly.